### PR TITLE
63 bring graphql types to frontend

### PR DESCRIPTION
**Pull request by EliasPinno**
Created the auto generated types from the GQL backend. First PR to migrate an actual request over is separate.

### CODE REVIEW

File: codegen.ts
Comment (EliasPinno) [inline codegen.ts:1]: The file which dictates where the codegen command looks to generate the types!

File: codegen.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type { CodegenConfig } from '@graphql-codegen/cli'
+ 
+const config: CodegenConfig = {
+   schema: 'https://iit-backend-v2.vercel.app/api/graphql',
```
Comment (EliasPinno) [inline codegen.ts:4]: Targeting the actual backend.

File: codegen.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+   overwrite: true,
+   documents: "./hooks/*.tsx",
```
Comment (EliasPinno) [inline codegen.ts:6]: Hooks is where all of our requests are!

File: gql/fragment-masking.ts
Comment (EliasPinno) [inline gql/fragment-masking.ts:1]: Autogen magic

File: gql/gql.ts
Comment (EliasPinno) [inline gql/gql.ts:1]: Autogen magic

File: gql/graphql.ts
Comment (EliasPinno) [inline gql/graphql.ts:1]: Actual types!

File: gql/json/schema.json
Comment (EliasPinno) [inline gql/json/schema.json:1]: Autgenerated schema json which tells us the observed types!

**Review by SeanKennyNF — APPROVED** [review]
Perfect, thanks man :)

**Pull request merged**